mlir/lib/Dialect/MemRef/Transforms/FoldMemRefAliasOps.cpp

```diff
@@ -63,39 +63,99 @@ resolveSourceIndicesExpandShape(Location loc, PatternRewriter &rewriter,
                                 memref::ExpandShapeOp expandShapeOp,
                                 ValueRange indices,
                                 SmallVectorImpl<Value> &sourceIndices) {
-  // The below implementation uses computeSuffixProduct method, which only
-  // allows int64_t values (i.e., static shape). Bail out if it has dynamic
-  // shapes.
-  if (!expandShapeOp.getResultType().hasStaticShape())
-    return failure();
-
+  // Record the rewriter context for constructing ops later.
   MLIRContext *ctx = rewriter.getContext();
+
+  // Record result type to get result dimensions for calulating suffix product
+  // later.
+  ShapedType resultType = expandShapeOp.getResultType();
+
+  // Traverse all reassociation groups to determine the appropriate indice
+  // corresponding to each one of them post op folding.
```

Comment:
I think all the logic here would be much simplified with https://github.com/llvm/llvm-project/commit/e095d978ba476c9624b4e72776089ea7301fa657 (it got reverted due to a bot failure, but should land soon enough). Maybe worth waiting for that?

Reply:
Do you think this change will break anything in that patch? (I don't see any changes in this pass in that patch)
Is that patch introducing better analysis utilities? (which can probably help us deal with dynamic and static cases in one go? - as you mentioned above)

I have no problem waiting for that change to get in if it somehow helps us better this patch.